### ISSUE TITLE
Fail to train TiTOK with VQGAN's training settings

### ISSUE THREAD

**Issue by ThisisBillhe** [bytedance/1d-tokenizer#19]
Hi, I use LlamaGen's [codebase](https://github.com/FoundationVision/LlamaGen) and can train VQGAN successfully. To train TiTOK, I just use a TiTokEncoder (just borrowed from your repo), a VectorQuantizer, a TiTokDecoder, and a PixelShuffleDecoder as you mentioned in another issue. The PixelShuffleDecoder consists of a linear layer to adjust num_channels and a nn.PixelShuffle to adjust the resolution. 

However, the perceptual loss decreases slowly and stops decreasing at a high value. The reconstructed image is also poor. Do you have any advice?

**Comment by MikeWangWZHL** [comment]
I also find it hard to reproduce the results; it would be great if the authors can lease the training code for both stage 1 and 2; thank you!

**Comment by cornettoyu** [comment]
Hi,

Thank you all for your interests in our work and patience. We have released the two-stage training code and feel free to check it.
